lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2806,100 +2811,133 @@ migrationPlanToSelectionWithdrawals plan rewardWithdrawal outputAddressesToCycle
+calcMinimumDeposit DBLayer{..}  netLayer wid =
+    throwInIO $ mapExceptT atomically (isStakeKeyRegistered wid) >>= \case
+        True -> pure $ Coin 0
+        False -> liftIO $ stakeKeyDeposit <$> currentProtocolParameters netLayer
   where
-    throwInIO :: ExceptT ErrNoSuchWallet IO a -> IO a
-    throwInIO x = runExceptT x >>= \case
-        Right a -> pure a
-        Left e -> throwIO $ ExceptionNoSuchWallet e
+    throwInIO :: forall a. ExceptT ErrNoSuchWallet IO a -> IO a
+    throwInIO = runExceptT >=> either (throwIO . ExceptionNoSuchWallet) pure
 
-    db = ctx ^. dbLayer @IO @s @k
-    nl = ctx ^. networkLayer
+data DelegationFee = DelegationFee
+    { feeSpread :: FeeSpread
+    , deposit :: Coin
+    }
 
--- | Estimate the transaction fee for a given coin selection algorithm by
--- repeatedly running it (100 times) and collecting the results. In the returned
--- 'FeeEstimation', the minimum fee is that which 90% of the sampled fees are
--- greater than. The maximum fee is the highest fee observed in the samples.
-estimateFee
+delegationFee
+    :: forall s k (n :: NetworkDiscriminant)
+     . ( AddressBookIso s
+       , s ~ SeqState n k
+       , BoundedAddressLength k
+       , Typeable k
+       , Typeable n
+       )
+    => DBLayer IO s k
+    -> NetworkLayer IO Read.Block
+    -> TransactionLayer k 'CredFromKeyK SealedTx
+    -> TimeInterpreter (ExceptT PastHorizonException IO)
+    -> AnyRecentEra
+    -> ChangeAddressGen s
+    -> WalletId
+    -> ExceptT ErrSelectAssets IO DelegationFee
+delegationFee db netLayer txLayer ti era changeAddressGen walletId = do
+    protocolParams <- liftIO $ currentProtocolParameters netLayer
+    WriteTx.withRecentEra era $ \(recentEra :: WriteTx.RecentEra recentEra) -> do
+        feeSpread <- calculateFeeSpread $ do
+            (Cardano.Tx (Cardano.TxBody bodyContent) _, _updatedWallet) <-
+                liftIO $ buildTransaction @s @k @n @recentEra
+                    db txLayer ti walletId changeAddressGen protocolParams
+                    defaultTransactionCtx []
+            pure $ case Cardano.txFee bodyContent of
+                Cardano.TxFeeExplicit _ coin -> Fee (fromCardanoLovelace coin)
+                Cardano.TxFeeImplicit Cardano.TxFeesImplicitInByronEra ->
+                    case recentEra of {}
+
+        deposit <- liftIO $ calcMinimumDeposit db netLayer walletId
+        pure DelegationFee {..}
+
+-- | For a given transaction fee estimation calculation evaluates its fee spread
+-- by repeatedly running it (100 times) and recording the results.
+-- In the returned 'FeeSpread', the minimum fee is that which 90% of the sampled
+-- fees are greater than. The maximum fee is the highest fee observed in the
+-- samples.
+calculateFeeSpread
     :: forall m. Monad m
-    => ExceptT ErrSelectAssets m Coin
-    -> ExceptT ErrSelectAssets m FeeEstimation
-estimateFee
+    => ExceptT ErrSelectAssets m Fee
+    -> ExceptT ErrSelectAssets m FeeSpread
```

Comment:
Ah, I take this `Fee -> FeeSpread/Estimation` is why you renamed it, but I'm not convinced as:
1. "FeeSpread" sounds like the _width_ of the distribution of fees, whereas we're also interested in the absolute values.
2. Why add the cognitive overhead of renaming functions we 1) want to remove when we are able to 2) now differ in naming compared to the endpoint handlers and documentation, which still use the "estimate fee" terminology? I don't think this makes sense.

Reply:
Changed to `(Percentile 10 Fee, Percentile 90 Fee)`